pkg/network/ebpf/c/protocols/classification/protocol-classification.h

```diff
@@ -218,19 +235,13 @@ __maybe_unused static __always_inline void protocol_classifier_entrypoint_tls_ha
     }
     tls_info_t* tls_info = get_tls_enhanced_tags(&classification_ctx->tuple);
     if (!tls_info) {
-        goto next_program;
+        return;
```

Comment:
Will this impact classification for other protocols like kafka that can also use TLS?

Reply:
these tail calls are only invoked directly when we know that the packet is a TLS handshake

Reply:
but it looks like regardless the USM stats are fairly stable: https://ddstaging.datadoghq.com/s/c894ecec5/zuk-x77-5ck

Reply:
I think by moving it from before to after the TLS inspection we are not messing with the classification of other protocols (but hoping Guy will chime in tomorrow)